services/qs-spark-service/src/main/java/ai/quantumics/api/adapter/AwsS3IamRoleConnection.java

```diff
@@ -0,0 +1,43 @@
+package ai.quantumics.api.adapter;
+
+import ai.quantumics.api.model.IamConnection;
+import ai.quantumics.api.req.AwsDatasourceRequest;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.google.gson.Gson;
+
+import static ai.quantumics.api.constants.DatasourceConstants.ROLE_ARN_SESSION_NAME;
+
+public class AwsS3IamRoleConnection implements AwsS3Connection{
+    @Override
+    public AmazonS3 getS3Connection(String connectionData, String cloudRegion) {
+        Gson gson = new Gson();
+        String roleArn = gson.fromJson(connectionData,IamConnection.class).getIam();
+        AWSCredentialsProvider credentialsProvider = getAwsCredentialsProvider(roleArn, cloudRegion);
+
+        // Create an Amazon S3 client using the assumed role's credentials.
+        return AmazonS3ClientBuilder.standard()
+                .withCredentials(credentialsProvider)
+                .withRegion(cloudRegion)
+                .build();
+    }
+
+    public AWSCredentialsProvider getAwsCredentialsProvider(String roleArn, String cloudRegion) {
+        int sessionDurationSeconds = 3600; // 1 hour
```

Comment:
where is sessionDurationSeconds is checked whether expired or not before creating new connection provider?